src/main/java/ladder/domain/Line.java

```diff
@@ -7,59 +7,69 @@
 import java.util.stream.IntStream;
 
 public class Line {
-    private final List<Boolean> line;
+    private final List<Rung> rungs;
 
-    public Line(int size, LineGenerator generator) {
-        this(createLine(size, generator));
+    public Line(int size, RungGenerator generator) {
+        this(generateRungs(size, generator));
     }
 
     public Line(boolean... line) {
-        this(createLine(line));
+        this(toRungs(line));
     }
 
-    public Line(List<Boolean> line) {
-        this.line = line;
+    public Line(List<Rung> rungs) {
+        this.rungs = rungs;
     }
 
-    private static List<Boolean> createLine(int size, LineGenerator generator) {
+    private static List<Rung> toRungs(boolean[] line) {
+        return IntStream.range(0, line.length)
+                .mapToObj(i -> new Rung(line[i]))
+                .collect(Collectors.toList());
+    }
+
+    private static List<Rung> generateRungs(int size, RungGenerator generator) {
         if (size < 1) {
             throw new IllegalArgumentException("Line size should be greater than 0");
         }
-        return createLine(generate(size, generator));
-    }
 
-    private static List<Boolean> createLine(boolean[] line) {
-        return IntStream.range(0, line.length)
-                .mapToObj(i -> line[i])
+        RungFactory factory = new RungFactory(generator);
+        return IntStream.range(0, size)
+                .mapToObj(i -> factory.createNextRung())
                 .collect(Collectors.toList());
```

Comment:
`RungFactory`를 보면 이전 rung 상태를 기록하기 위한 클래스같습니다.

여기서 Stream을 활용하면 더 쉽게 처리할 수 있을 것 같아요
```suggestion
        var initial = Rung.create(generator, null); // 초기값 생성
        return Stream.iterate(
                        initial,
                        previous -> Rung.create(generator, previous)
                )
                .limit(size)
                .collect(Collectors.toUnmodifiableList());
```
이러면 `RungFactory`없이도 누적된 rung 컬렉션을 생성할 수 있게됩니다!

첫번째 rung을 위해 null 입력 대신`Rung.create(generator)`을 추가로 만들어도 좋을 것 같네요 🚀